spinalcordtoolbox/registration/algorithms.py

```diff
@@ -1261,9 +1261,9 @@ def register2d(fname_src, fname_dest, fname_mask='', fname_warp='warp_forward.ni
         list_warp_inv = []
 
     # loop across slices
-    for i in range(nz):
+    print()  # Add newline between last log message and the progress bar logging
+    for i in sct_progress_bar(range(0, nz), unit='iter', unit_scale=False, desc="Registering slice", ncols=100):
```

Comment:
One note: For `register2d`, because we're calling an `ANTs` binary on every loop, we print out a very verbose message between each loop:

![image](https://user-images.githubusercontent.com/16181459/233430559-358cc4f5-b4e7-4fe6-8a35-a3744e82cfc6.png)

This happens both before and after these changes:

![image](https://user-images.githubusercontent.com/16181459/233430724-e5a2b298-bd5b-459b-8033-6bdec83f074a.png)

We could consider silencing these messages with `verbose=0`, which would instead get us this:

![image](https://user-images.githubusercontent.com/16181459/233431063-50b6178b-fc3e-4473-824d-0cf55d70adbb.png)

Much nicer! But, I wanted to bring this up before actually making the change.

Reply:
I think it's a good idea, but it would be nice to preserve debugging output if `verbose == 2`. So I added some commits to this PR to do just that.

Reply:
Thank you for cleaning up the verbose code! However, I think we need to set `verbose` in one extra place to solve this issue.

In https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4094/commits/05a0165362bc59c3d78731c6d83036edb7cad30d, `verbose` is set in the `cmd` list (which sets the verbosity value that gets passed to the ANTs binary itself.) However, `run_proc` has its own separate `verbose` parameter that needs to be set to suppress the blue messages about which command has been run.

I've added this in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4094/commits/722bf1cae126e7e0ed2aeb1c9e0305622e4a1637. If it looks good to you, feel free to merge this PR. :)